compiler/rustc_codegen_llvm/src/gotoc/cbmc/goto_program/symtab_transformer/gen_c_transformer/expr_transformer.rs

```diff
@@ -0,0 +1,277 @@
+    let right_mask = BigInt::from(u64::MAX);
+
+    let unsigned_typ = match typ {
+        Type::Unsignedbv { .. } => typ.clone(),
+        Type::Signedbv { width } => Type::unsigned_int(*width),
+        _ => panic!("Unexpected type in bignum_to_expr: {:?}", typ),
+    };
+    let left_half: BigInt = num.bitand(left_mask);
+    let right_half = num.bitand(right_mask);
+
+    Expr::int_constant(left_half.shr(64), unsigned_typ.clone())
+        .shl(Expr::int_constant(64, Type::c_int()))
+        .bitor(Expr::int_constant(right_half, unsigned_typ))
+        .cast_to(typ.clone())
+}
+
+/// Struct for handling the expression replacement transformations for --gen-c-runnable.
+pub struct ExprTransformer {
+    new_symbol_table: SymbolTable,
+    empty_statics: FxHashMap<String, Expr>,
+}
+
+impl ExprTransformer {
+    /// Replace expressions which lead to invalid C with alternatives.
+    pub fn transform(original_symbol_table: &SymbolTable) -> SymbolTable {
+        let new_symbol_table = SymbolTable::new(original_symbol_table.machine_model().clone());
+        ExprTransformer { new_symbol_table, empty_statics: FxHashMap::default() }
+            .transform_symbol_table(original_symbol_table)
+    }
+
+    /// Extract `empty_statics` map for final processing.
+    pub fn empty_statics_owned(&mut self) -> FxHashMap<String, Expr> {
+        std::mem::replace(&mut self.empty_statics, FxHashMap::default())
+    }
+
+    /// Add identifier to a transformed parameter if it's missing.
+    /// Necessary when function wasn't originally a definition.
+    fn add_parameter_identifier(&mut self, parameter: &Parameter) -> Parameter {
+        if parameter.identifier().is_some() {
+            parameter.clone()
+        } else {
+            let name = format!("__{}", type_to_string(parameter.typ()));
+            let parameter_sym = self.mut_symbol_table().ensure(&name, |_symtab, name| {
+                Symbol::variable(
+                    name.to_string(),
+                    name.to_string(),
+                    parameter.typ().clone(),
+                    Location::none(),
+                )
+            });
+            parameter_sym.to_function_parameter()
+        }
+    }
+}
+
+impl Transformer for ExprTransformer {
+    /// Get reference to symbol table.
+    fn symbol_table(&self) -> &SymbolTable {
+        &self.new_symbol_table
+    }
+
+    /// Get mutable reference to symbol table.
+    fn mut_symbol_table(&mut self) -> &mut SymbolTable {
+        &mut self.new_symbol_table
+    }
+
+    /// Get owned symbol table.
+    fn extract_symbol_table(self) -> SymbolTable {
+        self.new_symbol_table
+    }
+
+    /// Translate Implies into Or/Not.
+    fn transform_expr_bin_op(
+        &mut self,
+        _typ: &Type,
+        op: &BinaryOperand,
+        lhs: &Expr,
+        rhs: &Expr,
+    ) -> Expr {
+        let lhs = self.transform_expr(lhs);
+        let rhs = self.transform_expr(rhs);
+
+        match op {
```

Comment:
Make `binop` public, and this becomes a 2 liner :)

Reply:
Done.